playlist.c

```diff
@@ -2546,8 +2700,13 @@ bool playlist_init_cached(const playlist_config_t *config)
  **/
 playlist_t *playlist_init(const playlist_config_t *config)
 {
-   struct playlist_entry *entries = NULL;
-   playlist_t           *playlist = (playlist_t*)malloc(sizeof(*playlist));
+   struct playlist_entry *entries         = NULL;
+   playlist_t           *playlist         = (playlist_t*)malloc(sizeof(*playlist));
+   size_t i                               = 0;
+   size_t j                               = 0;
+   union string_list_elem_attr attributes = { 0 };
+   bool use_relative_path                 = !string_is_empty(config->base_content_directory);
+   char tmp_entry_path[PATH_MAX_LENGTH];
```

Comment:
Don't initialise variables you don't need here - much of this can go inside the `if (config->relative_paths)` condition